src/service.rs

```diff
@@ -1368,29 +1368,25 @@ impl Service {
 
     /// The equivalent of libp2p `inject_connected()` for a udp session. We have no stream, but a
     /// session key-pair has been negotiated.
-    fn inject_session_established(&mut self, enr: Enr, connection_update: ConnectionUpdate) {
+    fn inject_session_established(&mut self, enr: Enr, connection_direction: ConnectionDirection) {
         // Ignore sessions with non-contactable ENRs
         if self.ip_mode.get_contactable_addr(&enr).is_none() {
             return;
         }
 
         let node_id = enr.node_id();
 
-        let direction = match connection_update {
-            ConnectionUpdate::Incoming => ConnectionDirection::Incoming,
-            ConnectionUpdate::Outgoing => ConnectionDirection::Outgoing,
-            ConnectionUpdate::IncomingIfNotExists => {
-                let key = kbucket::Key::from(node_id);
-                match self
-                    .kbuckets
-                    .read()
-                    .get_bucket(&key)
-                    .map(|bucket| bucket.get(&key))
-                {
-                    Some(Some(node)) if node.status.is_connected() => node.status.direction,
```

Comment:
In this commit, I removed the `if node.status.is_connected()` condition because I think the condition gives attackers chances to promote their connection from Incoming to Outgoing.